specification/search/data-plane/Microsoft.Azure.Search.Service/readme.md

```diff
@@ -111,13 +111,136 @@ input-file:
 - track1/stable/2019-05-06/searchservice.json
 ```
 
+### Tag: track2-package-2019-05-preview
+
+These settings apply only when `--tag=track2-package-2019-05-preview` is specified on the command line.
+It is meant to be used for generating Track 2 Java only!
+
+``` yaml $(tag) == 'track2-package-2019-05-preview'
+input-file:
+- track2/preview/2019-05-06-preview/searchservice.json
+```
+
+### Tag: track2-package-2019-05
+
+These settings apply only when `--tag=track2-package-2019-05` is specified on the command line.
+It is meant to be used for generating Track 2 Java SDK only!
+
+``` yaml $(tag) == 'track2-package-2019-05'
+input-file:
+- track2/stable/2019-05-06/searchservice.json
+```
+
 ---
 # Code Generation
 
 !!! READ THIS !!!
-This swagger is not yet ready for languages other than C#.
+This swagger is ready for C# and Java.
 !!! READ THIS !!!
 
+## Java
+
+These settings apply only when `--java` is specified on the command line.
+Please also specify `--java-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(java)
+title: SearchServiceRestClient
+output-folder: $(java-sdks-folder)/search/Microsoft.Azure.Search.Service/src/Generated
+namespace: com.azure.search
+java: true
+sync-methods: none
+add-context-parameter: true
+generate-client-interfaces: false
+license-header: |-
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT License.
+  Code generated by Microsoft (R) AutoRest Code Generator.
+  Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+require: ./autorest-custom-directives.md
+
+vararg-properties: >-
+  CorsOptions.allowedOrigins
+
+directive:
+
+- rename-custom-property:
+    path: "$.parameters.ClientRequestIdParameter['x-ms-parameter-grouping']"
+    from: "search-request-options"
+    to: "request-options"
+
+# Add missing data types to DataType
+- from: swagger-document
+  where: $.definitions.DataType.enum
+  transform: |
+    if (!$.includes("Collection(Edm.String)")) {
+      $.push("Collection(Edm.String)",
+        "Collection(Edm.Int32)",
```

Comment:
We purposefully left these out of the enum since the collection-ness of a type ought to be orthogonal to the underlying type. In the .NET SDK for example, a string collection is defined as `DataType.Collection(DataType.String)`. It should work similarly in Java IMO.

Reply:
@brjohnstmsft 

1. Autorest does not generate these types.
2. It does not exists in Java

Therefore we had to manually modify and create it.

Reply:
If AutoRest doesn't generate this type, then I guess I'm unclear on why this transform is needed.

Reply:
yes, I meant to expose the hidden requirement

Reply:
@brjohnstmsft My understanding is the service (and in the portal) supports `Collection(Edm.Something)` but since it's not defined in the spec, AutoRest doesn't generate the corresponding code. Perhaps the spec needs to be updated so we can avoid adding this transform?

Reply:
@sakintoye The programming model we want for `DataType` is that collection types are not part of the enum -- they are constructed by a custom `Collection` function. See [the implementation in the .NET SDK](https://github.com/Azure/azure-sdk-for-net/blob/d02d5921a6c5518f785d214b202f2657a224f4cb/sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexes/Models/DataType.cs#L71) to see how this works.

AutoRest is never going to be able to generate that for us; The best it can do is generate an "extensible enum" for the base types -- I think even that doesn't work 100% yet, so `DataType` should be custom code at this point.

Reply:
A new User Story for converting DataType to custom code is opened here:
https://dev.azure.com/csee2e/Azure%20Search%20SDK/_workitems/edit/1717